src/background/Wallet/Wallet.ts

```diff
@@ -1320,6 +1361,31 @@ export class Wallet {
     });
   }
 
+  async signMessage({
+    signerAddress,
+    message,
+    messageContextParams,
+  }: {
+    signerAddress: string;
+    message: string;
+    messageContextParams: MessageContextParams;
+  }) {
+    const messageAsUtf8String = toUtf8String(message);
```

Comment:
All non-private methods should verify origin:

```suggestion
  async signMessage({ param: {
    signerAddress,
    message,
    messageContextParams,
  }, context }: WalletMethodParams<{
    signerAddress: string;
    message: string;
    messageContextParams: MessageContextParams;
  }>) {
    this.verifyInternalOrigin(context);
    const messageAsUtf8String = toUtf8String(message);
```